File: lib/utilities/market.dart
import 'package:namakala/utilities/product.dart';

class Market {
  late String name;
  List<Product> products = [];

  Market(this.name);

  Market.fromJson(Map<String, dynamic> json) {
    name = json['name'];
    for (Map<String, dynamic> p in json['products']) {
      products.add(Product.fromJson(p));
    }
  }

  Map<String, dynamic> toJson() => {
        'name': name,
      };
}

File: src/utilities/Market.java
package utilities;

import org.json.simple.JSONArray;
import org.json.simple.JSONObject;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class Market implements Serializable {
    private Person owner;
    private String name;
    private List<Product> products = new ArrayList<>();
    
    public Market(Person owner, String name) {
        this.owner = owner;
        this.name = name;
    }
    
    public JSONObject toJson() {
        JSONObject jo = new JSONObject();
        jo.put("owner", owner.toJson());
        jo.put("name", name);
    
        JSONArray ja = new JSONArray();
        for (Product p: products) {
            ja.add(p.toJson());
        }
        jo.put("products", ja);
        
        return jo;
    }
    
    public Person getOwner() {
        return owner;
    }
    
    public String getName() {
        return name;
    }
    
    public List<Product> getProducts() {
        return products;
    }
}
